agent/src/beerocks/slave/backhaul_manager/backhaul_manager_thread.h

```diff
@@ -47,6 +47,22 @@ class backhaul_manager : public btl::transport_socket_thread {
 
     // For agent_ucc_listener
     const std::string &get_controller_bridge_mac() { return controller_bridge_mac; }
+    /**
+     * @brief get radio mac (ruid) of registered slave based on frequency type
+     * 
+     * @param freq radio frequency to search
+     * @return radio mac of the found slave if found, otherwise empty string
+     */
+    const std::string freq_to_radio_mac(enum eFreqType freq) const;
+
+    /**
+     * @brief start WPS PBC
+     * 
+     * @param radio_mac radio mac of the radio on which to start WPS
+     * @param bssid bssid on which to start WPS
+     * @return true on success, false on failure
+     */
+    bool start_wps_pbc(const sMacAddr &radio_mac, const sMacAddr &bssid);
```

Comment:
(nitpick) it makes more sense to me to put the logic of which VAP to start WPS on *inside* this function.

And in keeping with my previous comment, the radio should be passed as a `const sRadioInfo&`.

Reply:
Refactored.